sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -87,8 +101,8 @@ case class CreateHiveTableAsSelectCommand(
       }
     } else {
       try {
-        sparkSession.sessionState.executePlan(InsertIntoTable(
-          metastoreRelation, Map(), query, overwrite = true, ifNotExists = false)).toRdd
+        sparkSession.sessionState.executePlan(InsertIntoTable(metastoreRelation,
```

Comment:
We changed the order, but are we able to handle the type difference? Are we doing casting? Do we have any test case to cover it?